### PR TITLE
Add Provider to Placement

### PR DESCRIPTION
**Pull request by Kizr**
## Context

School users should be able to select a provider for their placements then they know which provider will be assigned. This is an optional choice.

## Changes proposed in this pull request

- [x] Adds an optional `method` argument for the auto select form
- [x] Adds a new `name` delegation to `provider` for a Placement
- [x] Updates the Placement show page to display the provider and change link
- [x] Adds an edit page for editing providers on placements

## Guidance to review

- Create a new placement
- Click on the placement on the index page
- Click on Change in the provider row
- Select a provider
- Click on continue

## Link to Trello card

[Add Provider to Placement](https://trello.com/c/BSK7vNUk/269-provider-placements)

## Screenshots

image
image
image

### CODE REVIEW

File: app/views/placements/schools/placements/show.html.erb [hunk 1]
```diff
@@ -50,11 +50,16 @@
             <%= render Placement::StatusTagComponent.new(@placement.status) %>
           <% end %>
         <% end %>
+        <% summary_list.with_row do |row| %>
+          <% row.with_key(text: t(".attributes.placements.provider")) %>
+          <% row.with_value(text: @placement.provider_name.presence || t("placements.schools.placements.not_known_yet")) %>
```
Comment (JamieCleare2525) [inline app/views/placements/schools/placements/show.html.erb:55]: Might be worth moving this logic to the decorator.

File: app/controllers/placements/schools/placements_controller.rb [hunk 2]
```diff
@@ -12,6 +12,17 @@ def show; end
 
   def remove; end
 
+  def edit_provider
+    @providers = Provider.all
+  end
+
+  def update
+    @placement.provider = provider_params[:provider_name].present? ? Provider.find(provider_params[:provider_id]) : nil
```
Comment (JamieCleare2525) [inline app/controllers/placements/schools/placements_controller.rb:20]:
```suggestion
    @placement.provider = provider_params[:provider_name].present? && provider_params[:provider_id].present? ? Provider.find(provider_params[:provider_id]) : nil
```

Comment (Kizr) [reply app/controllers/placements/schools/placements_controller.rb:20]: The ID is funky here, the name is correct as it doesn't come through when deleting text. Looks wrong but it's correct.

I expect this to change/improve once designs come through

Comment (Kizr) [reply app/controllers/placements/schools/placements_controller.rb:20]: I realise I could still check the presence here, it's always present with the component but that doesn't stop me testing, I have made a similar change in the second PR I have up as I moved this logic.

**Review by JamieCleare2525 — APPROVED** [review]

**Review by Nitemaeric — APPROVED** [review]

**Pull request merged**